workspace/resource_notebook.go

```diff
@@ -101,8 +101,10 @@ var mtx = &sync.Mutex{}
 
 // Create creates a notebook given the content and path
 func (a NotebooksAPI) Create(r ImportPath) error {
-	mtx.Lock()
-	defer mtx.Unlock()
+	if r.Format == "DBC" {
+		mtx.Lock()
```

Comment:
can you switch to tree-based locking? this was added to prevent errors when creating notebooks along the tree

Reply:
DBC is kind of a specific case... I'm not sure how this will behave in the case of big trees, and/or in the case of multiple workspaces

Reply:
Hi Alex, thanks for the PR, a question -- what make case for .dbc files specific wrt to parallelism?

Reply:
DBC files may contain directories, so we need to make sure that the files that are potentially with directories, had a lock

Reply:
But I haven't seen DBC files deployed via TF in the real-life...

Reply:
From my reading of the comment next to the mutex, this applies to race conditions when creating/deleting directories, not notebooks. There should generally not be two notebook resources with the same path, otherwise there will always be a race condition. While it would be more precise to lock on a per-path basis, you would never get a diff-free plan after refreshing if two notebooks have the same path. So IMO this change is sufficiently safe.